datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -1169,21 +1281,73 @@ mod test {
     #[test]
     fn redundant_project_fields() {
         let table_scan = test_table_scan().unwrap();
+        let c_plus_a = col("c") + col("a");
+        let b_plus_a = col("b") + col("a");
         let expr_stats_1 = ExprStats::from([
-            ("c+a".to_string(), (1, DataType::UInt32)),
-            ("b+a".to_string(), (1, DataType::UInt32)),
+            (
+                Identifier {
+                    hash: 0,
+                    expr: &c_plus_a,
+                },
+                (1, DataType::UInt32),
+            ),
```

Comment:
I wonder if we could make these tests more readable with something like this or similar

```suggestion
            (
                Identifier::from(&c_plus_a).with_hash(0)
                (1, DataType::UInt32),
            ),
```

or

```suggestion
            (
                make_identifier(&c_plus_a, 0),
                (1, DataType::UInt32),
            ),
```

Reply:
Sure I can add `Identifier::from` to the PR.

Reply:
I forgot about these, will fix it a bit later today...

Reply:
Fixed in https://github.com/apache/datafusion/pull/10473/commits/a0913aa916df3c9904c33a225bbfedd6479eb1c1.